src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -513,6 +525,24 @@ internal override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArgu
             }
         }
 
+        private bool EmitNullablePublicOnly
+        {
+            get
+            {
+                var compilation = DeclaringCompilation;
+                return compilation.EmitNullablePublicOnly &&
+                    compilation.IsFeatureEnabled(MessageID.IDS_FeatureNullableReferenceTypes);
```

Comment:
>compilation.IsFeatureEnabled(MessageID.IDS_FeatureNullableReferenceTypes) [](start = 20, length = 73)

Having this property on the Module symbol as well feels confusing, especially that it has extra conditions. Effectively, there is an inconsistency between other users of compilation.EmitNullablePublicOnly and this code. Consider incorporating the feature check into the compilation's property and getting rid of this property.

Reply:
Alternatively, consider renaming this property to ```EmitNullablePublicOnlyAttribute```

---
In reply to: [293957455](https://github.com/dotnet/roslyn/pull/36398#discussion_r293957455) [](ancestors = 293957455)